snapshotting/manager.go

```diff
@@ -58,88 +53,68 @@ func NewSnapshotManager(baseFolder string) *SnapshotManager {
 	return manager
 }
 
-// AcquireSnapshot returns an idle snapshot if one is available for the given image
-func (mgr *SnapshotManager) AcquireSnapshot(image string) (*Snapshot, error) {
+// AcquireSnapshot returns a snapshot for the specified revision if it is available.
+func (mgr *SnapshotManager) AcquireSnapshot(revision string) (*Snapshot, error) {
 	mgr.Lock()
 	defer mgr.Unlock()
 
 	// Check if idle snapshot is available for the given image
-	idles, ok := mgr.idleSnapshots[image]
+	snap, ok := mgr.snapshots[revision]
 	if !ok {
-		mgr.idleSnapshots[image] = []*Snapshot{}
-		return nil, errors.New(fmt.Sprintf("There is no snapshot available for image %s", image))
+		return nil, errors.New(fmt.Sprintf("Get: Snapshot for revision %s does not exist", revision))
 	}
 
-	// Return snapshot for supplied image
-	if len(idles) != 0 {
-		snp := idles[0]
-		mgr.idleSnapshots[image] = idles[1:]
-		mgr.activeSnapshots[snp.GetId()] = snp
-		return snp, nil
+	// Snapshot registered in manager but creation not finished yet
+	if !snap.usable { // Could also wait until snapshot usable (trade-off)
```

Comment:
```suggestion
	if !snap.ready { // Could also wait until snapshot ready (trade-off)
```

Reply:
regarding the tradeoff... we should avoid failures due to quite arbitrary errors so if it's handled properly by the higher level logic, it's ok. a busy wait loop without a good reason can be more nasty because it can create long unexplainable delays

Reply:
I agree, dropped this comment.